frontend/src/components/BlocklyWorkspace/test-setup.ts

```
@@ -0,0 +1,4 @@
+import '@testing-library/jest-dom';
+
+// Mock setImmediate for our flushPromises utility
+(global as any).setImmediate = (callback: Function) => setTimeout(callback, 0);
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)